GoogleForCreators/web-stories-wp
# Taxonomies: Hookup Categories to backend

## Context

Taxonomies. [Designs](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=9721%3A383514)

## Summary

Hooks up hierachical input UI to the backend. Users can now add categories to their stories 🎉 

## Relevant Technical Choices

Added a lot of mocks here - so this pr looks a lot bigger than it actually is.

Some cleanup work was done too:
- add story response as a mock in `/db`
- fix some spelling errors

## To-do

- [x] ~make dropdown visible~ Add issue for dropdown https://github.com/google/web-stories-wp/issues/9049
- [x] add karma tests

## User-facing changes

<img src="https://user-images.githubusercontent.com/22185279/133659826-f70c943a-a34e-4db7-8cb5-8871ba2db0b9.gif" height="400px" /> <img src="https://user-images.githubusercontent.com/22185279/133660295-91ce2227-0a18-4e28-b0cd-227e97636c51.png" height="400px" />


## Testing Instructions






This PR can be tested by following these steps:

1. Enable the taxonomies feature flag
2. Click on the document tab in the inspector
3. Scroll to the categories component
4. Play with it! Should see:
    - Checkboxes can be checked and unchecked
    - Checkbox value won't persist unless you save as draft or publish your story
    - Should be able to add new categories and nest them (these persist on a page reload without saving the story)
    - Should be keyboard and screen reader accessible

## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #8841
Fixes #8840
Partially Addresses: #8837
